lib/src/message/abstract_message.dart

```
@@ -1,5 +1,208 @@
+import 'abort.dart';
+import 'abstract_message_with_payload.dart';
+import 'challenge.dart';
+import 'details.dart';
+import 'event.dart';
+import 'goodbye.dart';
+import 'invocation.dart';
+import 'message_types.dart';
+import 'published.dart';
+import 'registered.dart';
+import 'result.dart';
+import 'subscribed.dart';
+import 'unregistered.dart';
+import 'unsubscribed.dart';
+import 'welcome.dart';
+import 'error.dart';
+
 /// all WAMP messages have an id, this abstract message provides this default
 /// existing field
 abstract class AbstractMessage {
   late int id;
+
+  static AbstractMessage? fromJson(
+    Object? message,
+    AbstractMessageWithPayload Function(
+      AbstractMessageWithPayload message,
+      List<dynamic> messageData,
+      int argumentsOffset,
+    )
+        payloadInjector,
+  ) {
+    if (message is! List) {
```

Is this an error state? Should we be throwing exceptions rather than returning null?

---

The serializers returned null in original implementation so i didn't want to break it (or introduce a need to refactor all dependand components).